src/arkode/arkode_sprk.c

```diff
@@ -0,0 +1,574 @@
+  sprk_storage->ahat[3]          = SUN_RCONST(0.33400360328632142);
+  return sprk_storage;
+}
+
+ARKodeSPRKStorage ARKodeSymplecticMcLachlan5()
+{
+  ARKodeSPRKStorage sprk_storage = ARKodeSPRKStorage_Alloc(6);
+  sprk_storage->q                = 5;
+  sprk_storage->stages           = 6;
+  sprk_storage->a[0]             = SUN_RCONST(0.339839625839110000);
+  sprk_storage->a[1]             = -SUN_RCONST(0.088601336903027329);
+  sprk_storage->a[2]             = SUN_RCONST(0.5858564768259621188);
+  sprk_storage->a[3]             = -SUN_RCONST(0.603039356536491888);
+  sprk_storage->a[4]             = SUN_RCONST(0.3235807965546976394);
+  sprk_storage->a[5]             = SUN_RCONST(0.4423637942197494587);
+  sprk_storage->ahat[0]          = SUN_RCONST(0.1193900292875672758);
+  sprk_storage->ahat[1]          = SUN_RCONST(0.6989273703824752308);
+  sprk_storage->ahat[2]          = -SUN_RCONST(0.1713123582716007754);
+  sprk_storage->ahat[3]          = SUN_RCONST(0.4012695022513534480);
+  sprk_storage->ahat[4]          = SUN_RCONST(0.0107050818482359840);
+  sprk_storage->ahat[5]          = -SUN_RCONST(0.0589796254980311632);
+  return sprk_storage;
+}
+
+/*
+  The following methods are from:
+
+  Yoshida, H.: Construction of higher order symplectic integrators.
+  Phys Lett A. 150, 262–268 (1990).
+  https://doi.org/10.1016/0375-9601(90)90092-3
+
+ */
+
+ARKodeSPRKStorage ARKodeSymplecticYoshida6()
+{
+  ARKodeSPRKStorage sprk_storage = ARKodeSPRKStorage_Alloc(8);
+  sprk_storage->q                = 6;
+  sprk_storage->stages           = 8;
+  sprk_storage->a[0]             = SUN_RCONST(0.78451361047755726382);
+  sprk_storage->a[1]             = SUN_RCONST(0.23557321335935813368);
+  sprk_storage->a[2]             = -SUN_RCONST(1.17767998417887100695);
+  sprk_storage->a[3]             = SUN_RCONST(1.3151863206839);
+  sprk_storage->a[4]             = sprk_storage->a[2];
+  sprk_storage->a[5]             = sprk_storage->a[1];
+  sprk_storage->a[6]             = sprk_storage->a[0];
+  sprk_storage->a[7]             = SUN_RCONST(0.0);
+  sprk_storage->ahat[0]          = sprk_storage->a[0] / SUN_RCONST(2.0);
+  sprk_storage->ahat[1]          = (sprk_storage->a[0] + sprk_storage->a[1]) /
+                          SUN_RCONST(2.0);
+  sprk_storage->ahat[2] = (sprk_storage->a[1] + sprk_storage->a[2]) /
+                          SUN_RCONST(2.0);
+  sprk_storage->ahat[3] = (sprk_storage->a[2] + sprk_storage->a[3]) /
+                          SUN_RCONST(2.0);
+  sprk_storage->ahat[4] = sprk_storage->ahat[3];
+  sprk_storage->ahat[5] = sprk_storage->ahat[2];
+  sprk_storage->ahat[6] = sprk_storage->ahat[1];
+  sprk_storage->ahat[7] = sprk_storage->ahat[0];
+  return sprk_storage;
+}
+
+/*
+  The following methods are from:
+
+  McLachlan, R.I.: On the Numerical Integration of Ordinary Differential
+  Equations by Symmetric Composition Methods. Siam J Sci Comput. 16, 151–168
+  (1995). https://doi.org/10.1137/0916010
```

Comment:
According to Geometric Numerical Integration pg. 57, this method is originally due to M. Suzuki & K. Umeno https://doi.org/10.1007/978-3-642-78448-4_7

Reply:
Thanks! I'll add the reference and change the name in the code to SuzukiUmeno (if they found it first, I think the right thing to do is use their name).